junit-jupiter-api/src/main/java/org/junit/jupiter/api/DisplayNameGenerator.java

```diff
@@ -103,8 +134,38 @@ public interface DisplayNameGenerator {
 	 * @param testClass the class the test method is invoked on; never {@code null}
 	 * @param testMethod method to generate a display name for; never {@code null}
 	 * @return the display name for the test; never blank
+	 * @deprecated in favor of {@link #generateDisplayNameForMethod(List, Class, Method)}
 	 */
-	String generateDisplayNameForMethod(Class<?> testClass, Method testMethod);
+	@SuppressWarnings("DeprecatedIsStillUsed")
+	@API(status = DEPRECATED, since = "5.12")
+	@Deprecated
+	default String generateDisplayNameForMethod(@SuppressWarnings("unused") Class<?> testClass,
+			@SuppressWarnings("unused") Method testMethod) {
```

Comment:
Are the `@SuppressWarnings("unused")` declarations required only by IntelliJ IDEA?

If so, we don't need these.

Reply:
Removed